component/org.wso2.carbon.identity.dpop/src/main/java/org/wso2/carbon/identity/dpop/handler/DPoPAuthenticationHandler.java

```diff
@@ -0,0 +1,390 @@
+                    }
+                    try {
+                        String publicKey = getPublicKeyFromDpopProof(dpopHeader);
+                        if (StringUtils.isBlank(publicKey) ||
+                                !publicKey.equalsIgnoreCase(responseDTO.getTokenBinding().getBindingValue())) {
+                            return authenticationResult;
+                        }
+                    } catch (IdentityOAuth2Exception e) {
+                        return authenticationResult;
+                    }
+                } else {
+                    if (!authorizationHeader.startsWith(OAUTH_HEADER)) {
+                        return authenticationResult;
+                    }
+                }
+
+                if (!isTokenBindingValid(messageContext, binding,
+                        clientApplicationDTO.getConsumerKey(), accessToken)) {
+                    return authenticationResult;
+                }
+
+                authenticationResult.setAuthenticationStatus(AuthenticationStatus.SUCCESS);
+
+                if (StringUtils.isNotEmpty(responseDTO.getAuthorizedUser())) {
+                    User user = new User();
+                    String tenantAwareUsername =
+                            MultitenantUtils.getTenantAwareUsername(responseDTO.getAuthorizedUser());
+                    user.setUserName(UserCoreUtil.removeDomainFromName(tenantAwareUsername));
+                    user.setUserStoreDomain(UserCoreUtil.extractDomainFromName(tenantAwareUsername));
+                    user.setTenantDomain(MultitenantUtils.getTenantDomain(responseDTO.getAuthorizedUser()));
+                    authenticationContext.setUser(user);
+                }
+
+                authenticationContext.addParameter(CONSUMER_KEY, clientApplicationDTO.getConsumerKey());
+                authenticationContext.addParameter(OAUTH2_ALLOWED_SCOPES, responseDTO.getScope());
+                authenticationContext.addParameter(OAUTH2_VALIDATE_SCOPE,
+                        AuthConfigurationUtil.getInstance().isScopeValidationEnabled());
+                String serviceProvider = null;
+                try {
+                    serviceProvider =
+                            OAuth2Util.getServiceProvider(clientApplicationDTO.getConsumerKey()).getApplicationName();
+                } catch (IdentityOAuth2Exception e) {
+                    log.error("Error occurred while getting the Service Provider by Consumer key: "
+                            + clientApplicationDTO.getConsumerKey());
+                }
+
+                String serviceProviderTenantDomain = null;
+                try {
+                    serviceProviderTenantDomain =
+                            OAuth2Util.getTenantDomainOfOauthApp(clientApplicationDTO.getConsumerKey());
+                } catch (InvalidOAuthClientException | IdentityOAuth2Exception e) {
+                    log.error("Error occurred while getting the OAuth App tenantDomain by Consumer key: "
```

Comment:
Rethrow

Reply:
unable to rethrow since we couldn't change the overridden method signature.